libafl/src/schedulers/powersched.rs

```diff
@@ -211,6 +211,26 @@ where
         Ok(())
     }
 
+    fn on_replace(
```

Comment:
I'm not sure if powerschedule should have this method
you can change the metadata for each testcase here
but the global metadata (SchedulerMetadata) is not updated

See my comment here
https://github.com/AFLplusplus/LibAFL/pull/947#issuecomment-1373489957

Reply:
I reviewed the member of SchedulerMetadata.
So with the current state, it's possible to adjust it.

The values that need adjusting are `bitmap_size`, `bitmap_size_log`, and `exec_time`
so you can do like `scheduler_meta.bitmap_size = prev.bitmap_size + new.bitmap_size` for every member

Reply:
Btw could you add on_remove() too?

Reply:
Did change on on_replace myself

Reply:
AFAIK it should be basically the same for on_remove, no?

And, from my basic look, I believe these changes are correct. However, I don't have deep familiarity with power scheduling...

Reply:
I found an issue: we can't actually assume anything about the bitmap changing, since it's quite likely that the removed element did not actually reduce the bitmap size.

Reply:
Lol wait that's not right; the naming here is a bit misleading...
We would need to track the bitmap size as well, which is doable. Fixing.

Reply:
> it's quite likely that the removed/replaced element did not actually reduce the bitmap size.

yes, One corner case is these SchedulerTestcaseMetadata are not attached and SchedulerMetadata is unchanged.
(so they're removed __before__ calibration)

but even then all these minus values are 0 so they make no effect

Reply:
> the naming here is a bit misleading...
You can change to what you want 😄